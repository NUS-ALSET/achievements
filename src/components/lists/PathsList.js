/**
 * @file PathsList container module
 * @author Theodor Shaytanov <theodor.shaytanov@gmail.com>
 * @created 05.03.18
 */

import React from "react";
import PropTypes from "prop-types";

import { Link } from "react-router-dom";
import List, { ListItem, ListItemText, ListSubheader } from "material-ui/List";

import OpenInNewIcon from "material-ui-icons/OpenInNew";
import { pathSelect } from "../../containers/Paths/actions";

class PathsList extends React.PureComponent {
  static propTypes = {
    dispatch: PropTypes.func.isRequired,
    header: PropTypes.string.isRequired,
    paths: PropTypes.object,
    selectedPathId: PropTypes.string
  };

  selectPath = pathId => this.props.dispatch(pathSelect(pathId));

  render() {
    const { header, paths, selectedPathId } = this.props;

    return (
      <List
        component="nav"
        subheader={<ListSubheader component="div">{header}</ListSubheader>}
      >
        <ListItem
          onClick={() => this.selectPath("")}
          style={
            (!selectedPathId && {
              background: "rgba(0, 0, 0, 0.14)"
            }) ||
            {}
          }
        >
          <ListItemText inset primary="Default path" />
          <Link to="/paths">
            <OpenInNewIcon />
          </Link>
        </ListItem>
        {Object.keys(paths || {})
          .map(id => ({ ...paths[id], id }))
          .map(path => (
            <ListItem
              key={path.id}
              onClick={() => this.selectPath(path.id)}
              style={
                (selectedPathId === path.id && {
                  background: "rgba(0, 0, 0, 0.14)"
                }) ||
                {}
              }
            >
              <ListItemText inset primary={path.name} />
              <Link to="/paths">
                <OpenInNewIcon />
              </Link>
            </ListItem>
          ))}
      </List>
    );
  }
}

export default PathsList;
