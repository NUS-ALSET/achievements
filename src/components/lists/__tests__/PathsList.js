import React from "react";
import { StaticRouter } from "react-router";
import sinon from "sinon";
import { createShallow ,createMount} from "@material-ui/core/test-utils";
import PathsList from "../PathsList";

describe("<PathsList />", () => {

  let shallow,
  mount,
  mockDispatch;
  beforeEach(() => {
    shallow = createShallow();
    mount = createMount();
    mockDispatch=sinon.spy();
  });

  it("Should test PathsList component", () => {
    const props = {
        dispatch :()=>{},
        header :"",
        paths :{},
        selectedPathId :"",
        userId :""
    };
    const component = shallow(
      <PathsList {...props} />
    );
    expect(component).toMatchSnapshot();
  });

  it("Should test selectPath function of PathsList component", () => {
    const props = {
        dispatch :mockDispatch,
        header :"",
        paths :{},
        selectedPathId :"",
        userId :""
    };
    const component = shallow(
      <PathsList {...props} />
    );
    component.instance().selectPath("test");
    expect(mockDispatch.calledWith({
      type: "PATH_SELECT",
      pathId : "test"
    })).toEqual(true);
  });
});
