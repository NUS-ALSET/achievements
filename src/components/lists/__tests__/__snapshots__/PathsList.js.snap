// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PathsList /> Should test PathsList component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PathsList
    dispatch={[Function]}
    header=""
    paths={Object {}}
    selectedPathId=""
    userId=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <WithStyles(ListItem)
          onClick={[Function]}
          style={
                    Object {
                              "background": "rgba(0, 0, 0, 0.14)",
                            }
          }
>
          <WithStyles(ListItemText)
                    inset={true}
                    primary="Default path"
          />
          <Link
                    replace={false}
                    to="/paths/"
          >
                    <pure(OpenInNew) />
          </Link>
</WithStyles(ListItem)>,
        Array [],
      ],
      "component": "nav",
      "subheader": <WithStyles(ListSubheader)
        component="div"
/>,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(ListItemText)
              inset={true}
              primary="Default path"
/>,
            <Link
              replace={false}
              to="/paths/"
>
              <pure(OpenInNew) />
</Link>,
          ],
          "onClick": [Function],
          "style": Object {
            "background": "rgba(0, 0, 0, 0.14)",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "inset": true,
              "primary": "Default path",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <pure(OpenInNew) />,
              "replace": false,
              "to": "/paths/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(ListItem)
            onClick={[Function]}
            style={
                        Object {
                                    "background": "rgba(0, 0, 0, 0.14)",
                                  }
            }
>
            <WithStyles(ListItemText)
                        inset={true}
                        primary="Default path"
            />
            <Link
                        replace={false}
                        to="/paths/"
            >
                        <pure(OpenInNew) />
            </Link>
</WithStyles(ListItem)>,
          Array [],
        ],
        "component": "nav",
        "subheader": <WithStyles(ListSubheader)
          component="div"
/>,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(ListItemText)
                inset={true}
                primary="Default path"
/>,
              <Link
                replace={false}
                to="/paths/"
>
                <pure(OpenInNew) />
</Link>,
            ],
            "onClick": [Function],
            "style": Object {
              "background": "rgba(0, 0, 0, 0.14)",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "inset": true,
                "primary": "Default path",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <pure(OpenInNew) />,
                "replace": false,
                "to": "/paths/",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {},
  },
}
`;
